htsohm/htsohm.py

```diff
@@ -64,16 +64,21 @@ def htsohm(children_per_generation,    # number of materials per generation
            number_of_atomtypes,        # number of atom-types per material
            strength_0,                 # intial strength parameter
            number_of_bins,             # number of bins for analysis
-           max_generations=20):        # maximum number of generations
-
+           max_generations=20,         # maximum number of generations
+           acceptance_value=0.5):      # desired degree of `convergence`
     ############################################################################
     # write run-configuration file
     run_config = write_config_file(children_per_generation, number_of_atomtypes, strength_0,
         number_of_bins, max_generations)
     run_id = run_config["run-id"]
 
+    convergence = acceptance_value + 1          # initialize convergence with arbitrary value
     for generation in range(max_generations):
-        if generation == 0:                     # SEED GENERATION
-            seed_generation(run_id, children_per_generation, number_of_atomtypes)
-        elif generation >= 1:                   # FIRST GENERATION, AND ON...
-            next_generation(run_id, children_per_generation, generation)
+        while convergence >= acceptance_value:
+            if generation == 0:                     # SEED GENERATION
+                seed_generation(run_id, children_per_generation, number_of_atomtypes)
+                convergence = evaluate_convergence(run_id)
+            elif generation >= 1:                   # FIRST GENERATION, AND ON...
+                next_generation(run_id, children_per_generation, generation)
+                convergence = evaluate_convergence(run_id)
+            print('conergence:\t%s' % convergence)
```

Comment:
so if it converges, the above for loop will still spin through the remaining generations, but not do anything? Might make more sense to replace the while with just an if block like:

if convergence < accepted_value:
   break;

and then do the generations stuff normally